frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -185,13 +193,22 @@ export function DetailsGrid(props: DetailsGridProps) {
   const { sectionsFunc, resourceType, name, namespace, children, ...otherMainInfoSectionProps } =
     props;
   const location = useLocation<{ backLink: NavLinkProps['location'] }>();
+  const previousLocation = usePreviousRoute();
 
-  const backLink: string | undefined = React.useMemo(() => {
+  const backLink: string | Location | undefined = React.useMemo(() => {
```

Comment:
Just curious about useMemo with empty dependencies?

Reply:
I think this means the function inside will only be defined on component mount. Maybe it's not needed as the advantage we get is not much though maybe it protects for adding more heavy function logic in the future...

Reply:
I think we don't need the useMemo here cause a useMemo with empty array means it will be computed on every render, maybe in future when we have some extra optimization to be done then this makes more sense.

Reply:
That's not what the empty array does. An empty array means it only runs on mount and cleans up on unmount (i.e. runs just once). See the [notes](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects) in this section.
For the hook to run on every render, regardless of its dependencies, the array argument would have to be skipped/undefined.